src/radical/pilot/utils/component.py

```diff
@@ -4,301 +4,382 @@
 import copy
 import time
 import pprint
-import signal
 
-import setproctitle    as spt
 import threading       as mt
-import multiprocessing as mp
 import radical.utils   as ru
 
 from ..          import constants      as rpc
 from ..          import states         as rps
 
-from .queue      import Queue          as rpu_Queue
-from .queue      import QUEUE_OUTPUT   as rpu_QUEUE_OUTPUT
-from .queue      import QUEUE_INPUT    as rpu_QUEUE_INPUT
-from .queue      import QUEUE_BRIDGE   as rpu_QUEUE_BRIDGE
-
-from .pubsub     import Pubsub         as rpu_Pubsub
-from .pubsub     import PUBSUB_PUB     as rpu_PUBSUB_PUB
-from .pubsub     import PUBSUB_SUB     as rpu_PUBSUB_SUB
-from .pubsub     import PUBSUB_BRIDGE  as rpu_PUBSUB_BRIDGE
-
 
 # ------------------------------------------------------------------------------
 #
-class Component(ru.Process):
-    """
-    This class provides the basic structure for any RP component which operates
-    on stateful things.  It provides means to:
+class ComponentManager(object):
 
```

Comment:
We probably need a docstring here

Reply:
Well, it is not exposed to the end user, and I would not include it in the API documentation.  Having said that, it needs a docstring anyway for whoever reads that code - will do.